sympy/holonomic/holonomic.py

```diff
@@ -330,18 +314,16 @@ def __pow__(self, n):
 
         # if self is `Dx`
         if self.listofpoly == self.parent.derivative_operator.listofpoly:
-            sol = [self.parent.base.zero]*n
-            sol.append(self.parent.base.one)
+            sol = [self.parent.base.zero]*n + [self.parent.base.one]
             return DifferentialOperator(sol, self.parent)
 
         # the general case
-        else:
-            if n % 2 == 1:
-                powreduce = self**(n - 1)
-                return powreduce * self
-            elif n % 2 == 0:
-                powreduce = self**(n / 2)
-                return powreduce * powreduce
+        if n % 2 == 1:
+            powreduce = self**(n - 1)
+            return powreduce * self
+        elif n % 2 == 0:
+            powreduce = self**(n / 2)
+            return powreduce * powreduce
```

Comment:
```suggestion
        powreduce = self**(n // 2)
        return powreduce * powreduce
```

Can this be reduced if `n` is integer? (or if this function anyway doesn't work for fractional powers)
I also think that division should better be using exact quotient `//` because there could be possible bugs with precision.